syntax = "proto3";

// What does this service handle???
// 1. User Account
//      a. Create an account
//      b. Retrieve users account infos
//      c. Update any user info
//      d. Delete a user
// 2. Log a user in
// 3. Update a user password
// 4. Handle a password reset event

service User {
    rpc create_user (UserInfo) returns (Empty);
    rpc get_user_info (GetUser) returns (UserInfo);
    rpc modify_user (Empty) returns (Empty);
    rpc delete_user (Empty) returns (Empty);

    rpc login (LoginRequest) returns (Empty);
    rpc update_password (PasswordUpdate) returns (Empty);
}

message GetUser {
    string userID = 1;
}

message LoginRequest {
    string email = 1;
    string password = 2;
}

message UserInfo {
    // Login info
    string email = 1;
    string password = 2;
    // User info
    string firstname = 3;
    string lastname = 4;
    string birthday = 5;
    string language = 6;
}

message PasswordUpdate {
    string email = 1;
    string currentpassword = 2;
    string newpassword = 3;
}

message Empty {}

